### PR TITLE
fixing weights for PDF central values et. al.

### PR DESCRIPTION
**Pull request by tyjyang**
- fixed weights for PDF central values
- a minor variable scope bug for eta bin choice for muon scale unc
- a workaround for empty syst for nominal when writing combine datacard

Probably need a bit commit squashing, but just want to put it out here

### CODE REVIEW

File: wremnants/datasets/datagroups.py [hunk 1]
```diff
@@ -64,6 +64,7 @@ def setHists(self, baseName, syst, procsToRead=None, label=None, nominalIfMissin
             if selectSignal and group[label] and "signalOp" in group and group["signalOp"]:
                 group[label] = group["signalOp"](group[label])
         # Avoid situation where the nominal is read for all processes for this syst
+        if syst == "": foundExact = True
```
Comment (kdlong) [inline wremnants/datasets/datagroups.py:67]: I don't understand why this is needed. Shouldn't the readHist function work fine for syst="", and then foundExact is set to true in line 55? What do you run to produce an error?

Comment (tyjyang) [reply wremnants/datasets/datagroups.py]: I tested it without this line, the combine cards and input files were successfully produced, I will remove this line.

Comment (kdlong) [reply wremnants/datasets/datagroups.py]: Do you know what the issue was that caused you to run into this, though?

Comment (tyjyang) [reply wremnants/datasets/datagroups.py]: I couldn't recall the details now... should probably have kept a better log. I can't reproduce now what I saw, maybe other updates fixed this bug.

**Pull request closed without merging**